### ISSUE TITLE
Error on sending mail via msmtp

### ISSUE THREAD

**Issue by x-sven** [jjethwa/icinga2#212]
Hi, I've a running icinga2 container configured using msmtp like in this description: https://hub.docker.com/r/jordan/icinga2

Testing by: >echo "Hello this is sending email using msmtp" | msmtp *my-account*
works like expected. 

However, sending notifications results in the following error reported in /var/log/icinga2/icinga2.log:
[2019-12-05 06:15:29 +0000] warning/PluginNotificationTask: Notification command for checkable '(hidden address)!ssh-service-template' and notification '(hidden address)!ssh-service-template!Mail Service Notification' (PID: 20030, arguments: '/etc/icinga2/scripts/mail-host-notification.sh' '-4' '192.168.200.3' '-6' '' '-b' '' '-c' '' '-d' '2019-12-05 06:15:23 +0000' '-l' '***' '-n' '***' '-o' 'PING OK - Packet loss = 0%, RTA = 0.70 ms' '-r' 'admin@***' '-s''UP' '-t' 'PROBLEM') terminated with exit code 36, output: /etc/icinga2/scripts/mail-host-notification.sh: 148: [: unexpected operator
mail: cannot send message: Process exited with a non-zero status

Any ideas?

**Comment by jjethwa** [comment]
Hi @x-sven 

Have you tried executing the mail-host-notification.sh manually (with the arguments provided as an example) to see if it returns more information?

**Comment by x-sven** [comment]
Yes, that gives the same message. But, i've looked a little bit further... The problem I try to fix is, that I cannot send notifications from icinga2. The only hint I've found was that error in the icinga2.log. The line number, where this error is reported is just a debug output to the syslogger. I've commented that call - and the error message disappears. BUT, sending a notification does still not work. The msmtp.log states: errormsg='the server does not accept mail data'

Within the msmtp.log I've found out that the difference from using the "echo" cmd (above) is, that the FROM address is "root@icinga2docker". My webmail server will accept only a specific domain, let's assume  blabla.de. 

In my user.conf I changed the address of the user "icingaadmin" to a different address via:  email = "admin@blabla.de". Same I did in /etc/aliases: root:admin@blabla.de

At the moment, I cannot find the file e.g. to change the from-address that is used in the mail-host-notification.sh - script.

That error message caused by that verbose syslog call seems to be another problem...

**Comment by x-sven** [comment]
...latest updates: calling the  mail-host-notification.sh - script by adding '-f' 'admin@blabla.de' by hand works. It seems to me, that the MAILFROM argument is not used by the communication between icinga2 and the mail-host-notification.sh-script. msmtp thereafter does not replace the from (which is root@icinga2docker) based on the /etc/aliases this way. However, using msmtp the other way it does.

**Comment by jjethwa** [comment]
Hi @x-sven 

That's a good find!  I think it might be something that needs to be submitted upstream to the main icinga2 project so they can make sure MAILFROM is being passed to the script.  It might be something that is as easy as modifying the host notification object to send the parameter.  Please open one here: https://github.com/icinga/icinga2

**Comment by swtrustly** [comment]
Hi @x-sven 

Can you just verify that the `mail` command works in your container?
I believe it does, just to make sure.

````bash
mail -s "TEST" your@address.com
Cc:
Yada yada
Ctrl-D
````

**Comment by x-sven** [comment]
Hello @swtrustly,
 
> Can you just verify that the `mail` command works in your container?

Yes, I did, seems not to work this way :(
```
root@icinga2docker:/# mail -s "TEST" admin@blabla.de
Cc:
yada yada
cannot send message: Process exited with a non-zero status
root@icinga2docker:/#
```


However, this works: 
```
root@icinga2docker:/# echo "Hello this is sending email using msmtp" | msmtp root
```

*pling* - Mail received :)

Any suggestions how to find the reason for that? Google doesn't help so far...

**Comment by x-sven** [comment]
I've found something, that makes me nervous: 
cat /var/log/exim4/mainlog:

2019-12-10 16:50:08 1ieii8-0000Sb-7L <= root@icinga2docker U=root P=local S=378
2019-12-10 16:50:08 1ieii8-0000Sb-7L ** admin@blabla.de R=nonlocal: Mailing to remote domains not supported
2019-12-10 16:50:09 1ieii8-0000Sd-Se <= <> R=1ieii8-0000Sb-7L U=Debian-exim P=local S=1605
2019-12-10 16:50:09 1ieii8-0000Sd-Se ** admin@blabla.de <root@icinga2docker> R=nonlocal: Mailing to remote domains not supported
2019-12-10 16:50:09 1ieii8-0000Sd-Se Frozen (delivery error message)
2019-12-10 16:50:10 1ieii8-0000Sb-7L Completed

**Comment by x-sven** [comment]
...ok, from what I've found at google, the mail command uses the exim4 mechanism to send mail. The configuration in the image seems to set to local mail only. This makes sense, since msmtp is used to deliver/ forward mail. If I change the exim configuration by: dpkg-reconfigure exim4-config, I will be ask for smpt details of the server I want the mail forward to. At the moment I don't think that mail is a "must have" since msmtp works. Is that assumption right?

**Comment by jjethwa** [comment]
@x-sven That's correct, msmtp is the way to go with the container

**Comment by swtrustly** [comment]
`mail` should work though, it is a wrapper to `sendmail` that links to `msmtp`. This is probably why your notification script is funky. 
Can you trace backwards? We know `msmtp` works, can you try to send with `sendmail`?

**Comment by x-sven** [comment]
`echo "Subject: sendmail test" | sendmail admin@blabla.de`
...is working.

**Comment by x-sven** [comment]
I've set the field "notification_from" in the template.conf within /etc/icinga2. This relates to the manual/ "old" way to edit the config. I'm eager to set the notifications up by using the icinga2 director, however, I could still not get it working so far. So, setting up this variable and restarting the icinga2 service in the container results in getting mails, as if there was never a problem before...  

Now, after finding that, I was looking how to set those variables within the director. It works in the similar way, but was hard to find out (for me): 
Icinga Director -> Define Data Fields -> +Add
Field name = notification_from
Data type = String

Then,
Icinga Director -> Notifications -> Notification templates
Edit the proper templates, by selecting the tab Fields and add the former defined field, better "Mandatory" to be sure... Fill in the proper address in the Custom properties field, that appears notification template after storing the field. Deploy the config and notifications are working even without the field "notification_from" in the template.conf from before. 

ABSOLUTELY EVIL: 
The error message within the icinga2.log is still there! Therefore, the error message has nothing to do with a working e-mail notification setup. (OK: that's my current working hypothesis, but I've got a hundred mails today from the docker container :o)
